### PR TITLE
[1.0.1] P2P: syncing fix

### LINKED ISSUE #670: P2P syncing re-requesting ranges
Noticed syncing up to head of Jungle.

```
info  2024-08-29T12:37:25.614 net-2     net_plugin.cpp:2155           operator()           ] ["eosn-jungle4-seed171:9876 - 0599f3a" - 5 209.249.216.166:9876]   requesting range 157618277 to 157623276, fhead 157618279, lib 157618276
info  2024-08-29T12:37:28.161 net-2     net_plugin.cpp:2155           operator()           ] ["89.38.98.37:9876 - cbb90ba" - 1 89.38.98.37:9876]                requesting range 157623277 to 157628276, fhead 157618946, lib 157618944
info  2024-08-29T12:37:36.736 net-1     net_plugin.cpp:2155           operator()           ] ["89.38.98.37:9876 - cbb90ba" - 1 89.38.98.37:9876]                requesting range 157621276 to 157626275, fhead 157621278, lib 157621275
info  2024-08-29T12:37:39.542 net-6     net_plugin.cpp:2155           operator()           ] ["perth-jungle-pn-1:9876 - 583d13a" - 8 14.200.208.8:9876]         requesting range 157626276 to 157631275, fhead 157622010, lib 157622008
info  2024-08-29T12:37:48.697 net-5     net_plugin.cpp:2155           operator()           ] ["perth-jungle-pn-1:9876 - 583d13a" - 8 14.200.208.8:9876]         requesting range 157624482 to 157629481, fhead 157624483, lib 157624481
```

### PR DESCRIPTION
**Pull request by heifner**
Two small P2P syncing fixes:
- Do not request next range of blocks if `sync_last_requested_num` is 0 as that case is handled in the first part of the `if`. If we do request a new range, request from a potentially different peer. Noticed that in practice it would often stick to its current peer here as this would be the first time it realized it needs a new range of blocks.
- Update `sync_active_time` when a block is received. Noticed this "timer" was firing simply because it was taking too long to apply the blocks. I was testing with `sync-fetch-span` of 5000.
- Normalize on using `steady_clock` for stopwatch functionality.

Note: May want to re-target this to `main`.

Resolves #670

### CODE REVIEW

File: plugins/net_plugin/net_plugin.cpp [hunk 1]
```diff
@@ -2485,6 +2483,7 @@ namespace eosio {
          return;
       }
       c->latest_blk_time = std::chrono::system_clock::now();
+      sync_active_time = std::chrono::steady_clock::now(); // reset when we receive a block
```
Comment (linh2931) [inline plugins/net_plugin/net_plugin.cpp:2486]: You can do `sync_active_time = c->latest_blk_time = std::chrono::system_clock::now();` to guarantee they are all the same.

**Review by linh2931 — APPROVED** [review]

**Review by greg7mdp — APPROVED** [review]

**Comment by greg7mdp** [comment]
Looks like `latest_msg_time` should use `steady_clock` as well.

**Review by greg7mdp — APPROVED** [review]

**Comment by ericpassmore** [comment]
Note:start
group: STABILITY
category: INTERNALS
summary: Peer syncing fixes to improve active sync time, better normalizer timer, and skip unneeded request for additional blocks.
Note:end

**Comment by ericpassmore** [comment]
Note:start
category: Other
component: Plugins/Cleos/Utils
summary: Peer syncing fixes to improve active sync time, better normalizer timer, and skip unneeded request for additional blocks.
Note:end

**Pull request merged**